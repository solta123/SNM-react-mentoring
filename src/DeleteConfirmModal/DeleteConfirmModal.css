.ConfirmButton {
    float: right;
}

.DeleteConfirmText {
    padding: 0.5rem 0;
}

.DeleteConfirmCloseIcon {
    position: absolute;
    right: 1.5rem;
    top: 1.5rem;
    cursor: pointer;
}