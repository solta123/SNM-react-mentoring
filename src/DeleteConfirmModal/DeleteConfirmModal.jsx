import './DeleteConfirmModal.css';
import React from 'react';
import { Button, Card, Typography } from '@material-ui/core';
import CloseIcon from '@material-ui/icons/Close';

const DeleteConfirmModal = React.forwardRef((props, ref) => {


    return <Card className="ModalContainerCard" ref={ref}>
        <CloseIcon className="DeleteConfirmCloseIcon" onClick={props.onClose} />
        <Typography variant="h4">Delete {props.type}</Typography>
        <Typography className="DeleteConfirmText">Are you sure you want to delete this {props.type}: {props.itemName}?</Typography>
        <Button variant="contained" color="primary" className="ConfirmButton" onClick={props.onConfirm}>Confirm</Button>
    </Card>
});

export default DeleteConfirmModal;