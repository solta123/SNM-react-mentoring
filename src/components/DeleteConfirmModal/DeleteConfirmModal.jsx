import './DeleteConfirmModal.css';
import React from 'react';
import { Button, Card, Typography } from '@material-ui/core';
import CloseIcon from '@material-ui/icons/Close';
import PropTypes from 'prop-types';
import { useTranslation } from "react-i18next";

const DeleteConfirmModal = React.forwardRef((props, ref) => {
    // eslint-disable-next-line no-unused-vars
    const { t, i18n } = useTranslation('common');

    return <Card className="ModalContainerCard" ref={ref}>
        <CloseIcon className="DeleteConfirmCloseIcon" onClick={props.onClose} />
        <Typography variant="h4">{t('delete_type', { type: t(props.type + '_with_object_affix') })}</Typography>
        <Typography className="DeleteConfirmText">{t('are_you_sure_to_delete', { type: t(props.type + '_with_object_affix') })}: {props.itemName}?</Typography>
        <Button variant="contained" color="primary" className="ConfirmButton" onClick={props.onConfirm}>{t('confirm')}</Button>
    </Card>
});

DeleteConfirmModal.displayName = 'DeleteConfirmModal';

DeleteConfirmModal.propTypes = {
    onClose: PropTypes.func,
    type: PropTypes.string,
    itemName: PropTypes.string,
    onConfirm: PropTypes.func
}

export default DeleteConfirmModal;