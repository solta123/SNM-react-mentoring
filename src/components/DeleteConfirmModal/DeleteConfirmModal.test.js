import React, { Suspense } from 'react';
import renderer from 'react-test-renderer';
import DeleteConfirmModal from './DeleteConfirmModal';
import { I18nextProvider } from "react-i18next";
import i18next from "i18next";

describe('DeleteConfirmModal', () => {
    it('should render correctly with inputs', () => {
        const tree = renderer.create(
            <Suspense fallback="loading">
                <I18nextProvider i18n={i18next}>
                    <DeleteConfirmModal type={'movie'} itemName={'Star Wars'} />
                </I18nextProvider>
            </Suspense>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
});