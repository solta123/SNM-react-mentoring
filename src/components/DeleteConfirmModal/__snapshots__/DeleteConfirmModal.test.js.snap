// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteConfirmModal should render correctly with inputs 1`] = `"loading"`;
